nixos/modules/services/web-apps/spliit.nix

```diff
@@ -0,0 +1,177 @@
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "spliit";
+      description = mdDoc "Group under which Spliit runs.";
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = mdDoc "Whether to open the firewall for the specified port.";
+    };
+
+    configureNginx = mkOption {
+      type = types.bool;
+      default = true;
+      description = "Configure nginx as a reverse proxy for Spliit.";
+    };
+
+    host = mkOption {
+      type = lib.types.str;
+      description = "Domain on which nginx will serve Spliit";
+    };
+
+    database = {
+      createLocally = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Create the PostgreSQL database and database user locally.
+        '';
+      };
+      hostname = mkOption {
+        type = types.str;
+        default = "/run/postgresql";
+        description = "Database hostname.";
+      };
+      port = mkOption {
+        type = types.port;
+        default = 5432;
+        description = "Database port.";
+      };
+      name = mkOption {
+        type = types.str;
+        default = "spliit";
+        description = "Database name.";
+      };
+      user = mkOption {
+        type = types.str;
+        default = "spliit";
+        description = "Database user.";
+      };
+      passwordFile = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/run/keys/spliit-dbpassword";
+        description = ''
+          A file containing the password corresponding to
+          [](#opt-services.spliit.database.user).
+        '';
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.spliit = {
+      description = "Spliit bill-splitting web application";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ] ++ lib.optional cfg.database.createLocally "postgresql.service";
+      requires = lib.mkIf cfg.database.createLocally [ "postgresql.service" ];
+
+      environment = lib.mkMerge [
+        (lib.mkIf cfg.database.createLocally {
+          POSTGRES_PRISMA_URL = "postgresql://${cfg.database.user}@${cfg.database.name}/${cfg.database.name}?host=/var/run/postgresql/";
+          POSTGRES_URL_NON_POOLING = "postgresql://${cfg.database.user}@${cfg.database.name}/${cfg.database.name}?host=/var/run/postgresql/";
+        })
+        {
+          PORT = toString cfg.port;
+        }
+      ];
+
+      preStart = lib.mkIf (cfg.database.passwordFile != null) ''
+        DATABASE_PASSWORD="$(cat ${cfg.database.passwordFile})"
+
+        POSTGRES_PRISMA_URL="postgresql://${cfg.database.user}:$DATABASE_PASSWORD@${cfg.database.name}?host=${cfg.database.hostname}:${toString cfg.database.port}";
+
+        echo -n "POSTGRES_PRISMA_URL=\"$POSTGRES_PRISMA_URL\"" > /run/spliit/.env
+        echo -n "POSTGRES_URL_NON_POOLING=\"$POSTGRES_PRISMA_URL\"" > /run/spliit/.env
```

Comment:
Do you have a idea, how the POSTGRES_PRISMA_URL variables can be given to the service when using cfg.database.passwordFile.

I'm out of ideas.

Reply:
Seems like this is not supported by prisma (https://github.com/sfackler/rust-postgres/issues/729).

If there is no feasible way to construct the env variables from both the password file and the options, the user would have to enter the whole connection string in the password file and not use the other cfg.database.* parameters.

Reply:
Thanks, that seems reasonable.